workflows/prognostic_c48_run/runtime/loop.py

```diff
@@ -85,6 +86,44 @@ def get_momentum_diagnostics(self, state, tendency) -> Diagnostics:
         return {}
 
 
+class TendencyScalingStepper:
+    """Stepper interface
+
+    Steppers know the difference between tendencies, diagnostics, and
+    in-place state updates, but they do not know how and when these updates
+    will be applied.
+
+    Note:
+        Uses typing_extensions.Protocol to avoid the need for explicit sub-typing
+
+    """
+
+    def __init__(self, stepper: Stepper, scales: Mapping[str, float]):
+        self._stepper = stepper
+        self._scales = scales
+
+    @property
+    def label(self) -> str:
+        """Label used for naming diagnostics.
+        """
+        return self._stepper.label
+
+    def __call__(self, time, state) -> Tuple[Tendencies, Diagnostics, State]:
+        tendencies, diagnostics, state = self._stepper(time, state)
+        for name, scale in self._scales.items():
+            if name in tendencies:
+                tendencies[name] *= scale
+        return tendencies, diagnostics, state
```

Comment:
What if there is some item in `diagnostics` that is computed from something in `tendencies`? (like a column-integrated tendency). This would lead to an incorrect diagnostic.